interface/patient_file/summary/demographics.php

```diff
@@ -1038,9 +1047,9 @@ function setMyPatient() {
                     <?php
                     $t = $twig->getTwig();
 
-                    $allergy = (AclMain::aclCheckIssue('allergy')) ? 1 : 0;
-                    $pl = (AclMain::aclCheckIssue('medical_problem')) ? 1 : 0;
-                    $meds = (AclMain::aclCheckIssue('medication')) ? 1 : 0;
+                    $allergy = (AclMain::aclCheckIssue('allergy') ? 1 : 0) && !in_array('card_allergies', $hiddenCards) ? 1 : 0;
```

Comment:
won't this result in  boolean instead of 1 or 0. Wouldn't this make more sense:
```
$allergy = (AclMain::aclCheckIssue('allergy') && !in_array('card_allergies', $hiddenCards)) ? 1 : 0;
```